RecoParticleFlow/PFClusterProducer/interface/InitialClusteringStepBase.h

```diff
@@ -43,19 +44,34 @@ class InitialClusteringStepBase {
     conf.getParameterSetVector("thresholdsByDetector");
     for( const auto& pset : thresholds ) {
       const std::string& det = pset.getParameter<std::string>("detector");
-      const double& thresh_E = 
-	pset.getParameter<double>("gatheringThreshold");
-      const double& thresh_pT = 
-	pset.getParameter<double>("gatheringThresholdPt");
-      const double thresh_pT2 = thresh_pT*thresh_pT;
+
+      std::vector<int> depths;
+      std::vector<double> thresh_E;
+      std::vector<double> thresh_pT ;
+      std::vector<double> thresh_pT2;
+
+      if (det==std::string("HCAL_BARREL1") || det==std::string("HCAL_ENDCAP")) {
+	depths = pset.getParameter<std::vector<int> >("depths");
+	thresh_E = pset.getParameter<std::vector<double> >("gatheringThreshold");
+	thresh_pT = pset.getParameter<std::vector<double> >("gatheringThresholdPt");
```

Comment:
please add a check and throw an exception in a case that the vector sizes do not match